technical/GITHUB_ACTIONS.md

```diff
@@ -127,25 +131,35 @@ jobs:
           path: ~/.sonar/cache
           key: ${{ runner.os }}-sonar
           restore-keys: ${{ runner.os }}-sonar
-      - name: Cache Gradle packages
+      - name: Cache Maven packages
         uses: actions/cache@v1
         with:
-          path: ~/.gradle/caches
-          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle') }}
-          restore-keys: ${{ runner.os }}-gradle
+          path: ~/.m2
+          key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+          restore-keys: ${{ runner.os }}-m2
+      - name: Create custom Maven Settings.xml #(2)
+        uses: whelk-io/maven-settings-xml-action@v18
+        with:
+          output_file: custom_maven_settings.xml
+          servers: '[{ "id": "github-packages-compas", "username": "OWNER", "password": "${{ secrets.GITHUB_TOKEN }}" }]'
       - name: Build and analyze
         env:
-          GITHUB_USERNAME: "OWNER" #(2)
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} #(3)
+          GITHUB_USERNAME: "OWNER"
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
-        run: ./gradlew build jacocoTestReport sonarqube -Dsonar.projectKey=<insert project key> -Dsonar.organization=com-pas -Dsonar.host.url=https://sonarcloud.io --info #(4)
+        run: | #(3)
+          mvn -s custom_maven_settings.xml -B -Psonar \
+          -Dsonar.projectKey=<insert project key> \
+          -Dsonar.organization=com-pas \
+          -Dsonar.host.url=https://sonarcloud.io \
+          verify org.sonarsource.scanner.maven:sonar-maven-plugin:sonar
 ```
 
 A few points to remember:
 - (1): By default, all actions are triggered on a push action.
-- (2): Only applicable if your repository is depending on our [Github Package](https://github.com/orgs/com-pas/packages). The GITHUB_USERNAME needs to be set with something (value doesn't matter) to give your access to the Github Packages during build.
-- (3): Again, only applicable if your repository is depending on our Github Packages. The GITHUB_TOKEN gives you access to the Github Packages during build.
-- (4): Replace the `<insert project key>` with the project key you copied.
+- (2): Only applicable if your repository is depending on our [Github Package](https://github.com/orgs/com-pas/packages).
+  We need a custom `settings.xml` having the credentials. For more information, check our [Contributing](https://github.com/com-pas/contributing/blob/master/CONTRIBUTING.md).
```

Comment:
Maybe always just add this steps, otherwise you also need to explain that the option "-s custom_maven_settings.xml" needs to be removed.